### PR TITLE
Add error handling for store operations in cache/reflector

### PR DESCRIPTION
**Pull request by nvartolomei**
In my opinion the errors there should be returned, but to have minimal impact I just added runtime error handler.

Let me know what you think.

### CODE REVIEW

**Comment by k8s-oncall** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37688)

File: pkg/client/cache/reflector.go [hunk 1]
```diff
@@ -363,14 +363,23 @@ loop:
 			newResourceVersion := meta.GetResourceVersion()
 			switch event.Type {
 			case watch.Added:
-				r.store.Add(event.Object)
+				err := r.store.Add(event.Object)
+				if err != nil {
+					utilruntime.HandleError(fmt.Errorf("%s: unable to add watch event object to store %#v", r.name, event))
```
Comment (ncdc) [inline pkg/client/cache/reflector.go]: I'd recommend displaying err as well

**Comment by caesarxuchao** [comment]
lgtm after addressing @ncdc's comment. Thanks.

**Comment by nvartolomei** [comment]
@caesarxuchao amended required changes.

Please consider one more time returning errors there. The whole responsibility of a reflector is to keep store in sync, and if it unable to do so, imho, it should error out and let the creator of reflector handle the error.

**Comment by caesarxuchao** [comment]
@k8s-bot ok to test

**Comment by caesarxuchao** [comment]
@nvartolomei If the reflector returns the error to a controller, what can the controller do? I think in most cases a controller will log the error and restart the reflector, that's no different than the current implementation.

File: pkg/client/cache/reflector.go [hunk 2]
```diff
@@ -363,14 +363,23 @@ loop:
 			newResourceVersion := meta.GetResourceVersion()
 			switch event.Type {
 			case watch.Added:
-				r.store.Add(event.Object)
+				err := r.store.Add(event.Object)
+				if err != nil {
+					utilruntime.HandleError(fmt.Errorf("%s: unable to add watch event object (%#v) to store: %v", r.name, event, err))
```
Comment (caesarxuchao) [inline pkg/client/cache/reflector.go]: s/event/event.Object?

**Comment by caesarxuchao** [comment]
lgtm. Thanks!

**Comment by ncdc** [comment]
@k8s-bot cvm gce e2e test this issue: #26955

**Pull request merged**